src/models/connectionCredentials.ts

```diff
@@ -281,6 +281,14 @@ export class ConnectionCredentials implements IConnectionCredentials {
         return authenticationType === utils.authTypeToString(AuthenticationTypes.SqlLogin);
     }
 
+    public static isPasswordBasedConnectionString(connectionString: string): boolean {
+        return connectionString.toLowerCase().includes('password');
```

Comment:
Similar to the connectionString check below - password is something I could easily see a user having in a database/server name. This is a bit trickier to solve though - I don't know offhand if there's any libraries we could use for this but I would suggest looking around to see how others are solving this problem.